packages/web3/src/entitlement.ts

```diff
@@ -123,6 +122,8 @@ export const NoopRuleData = {
 type EntitledWalletOrZeroAddress = string
 
 export type LogicalOperation = OrOperation | AndOperation
+export type SupportedLogicalOperationType = LogicalOperation['logicalType']
+
 export type Operation = CheckOperation | OrOperation | AndOperation | NoOperation
```

Comment:
Does it make sense to change this line to 

`export type Operation = CheckOperation | LogicalOperation | NoOperation`
so that we don't have to repeat `AndOperation | OrOperation` anywhere?

Reply:
good suggestion, i'll try that and see how it looks

Reply:
I looked into it and I'm going to call this a bad suggestion because it logically breaks the `isLogicalOperation` type guard which converts objects to LogicalOperation, since NoOperation is not a logical operation (it doesn't have a left and right op, and doesn't work in the tree)

The existing code doesn't repeat `AndOperation | OrOperation` anywhere so I don't think this is an issue.